build_projects/shared-build-targets-utils/Utils/DotNetCli.cs

```diff
@@ -4,20 +4,22 @@
 using System.Runtime.InteropServices;
 using Microsoft.DotNet.Cli.Build.Framework;
 using Microsoft.DotNet.InternalAbstractions;
+using System.Collections.Generic;
 
 namespace Microsoft.DotNet.Cli.Build
 {
-    public class DotNetCli
+    public partial class DotNetCli
     {
         public static readonly DotNetCli Stage0 = new DotNetCli(GetStage0Path());
-        public static readonly DotNetCli Stage1 = new DotNetCli(Dirs.Stage1);
-        public static readonly DotNetCli Stage2 = new DotNetCli(Dirs.Stage2);
 
         public string BinPath { get; }
+        public IEnumerable<string> SharedFxVersionDirectories { get; private set; }
```

Comment:
My general design philosophy is "don't expose things that don't need to be exposed".  "Because someone might need it later" isn't a valid reason. In technical aspects: you make the API bloated, the object is larger than it has to be, and if you ever want to refactor it later - you need to check if someone is already using it since it is public.

By not making the property in the first place, you eliminate all these problems.  "If someone needs it later", they are totally free to come in and add the property later. But that will only happen when it is needed, not "just in case".